src/js/lib/routing.ts

```diff
@@ -0,0 +1,58 @@
+import { toggleCardDetails } from './utils';
+
+/**
+ * Pushes the card ID to the history stack. Prevents the page from jumping
+ * as it would happen when setting the location.hash.
+ * @param {string} id - The ID of the card
+ */
+export function pushCardState(id: string): void {
+  history.pushState({ id }, null, `#${id}`);
```

Comment:
This currently doesn't play nice with filters; if an item is selected and then a filter is applied, the hash gets removed. If a filter is applied first, the hash gets added at the end as a value for a query parameter.

Reply:
What would you suggest? Removing the hash altogether when filtering? Keeping the hash only if the card is in the filtered set of cards?